api-controllers/ServiceFabrikAdminController.js

```diff
@@ -52,19 +68,41 @@ class ServiceFabrikAdminController extends FabrikBaseController {
           });
         });
     }
-    logger.info(`Forbidden Manifest flag set to ${allowForbiddenManifestChanges}`);
-    if (allowForbiddenManifestChanges === false) {
-      const instanceId = this.parseServiceInstanceIdFromDeployment(deploymentName);
-      return this
-        .getServicePlanIdForInstanceId(instanceId)
-        .then(planId => this.getOutdatedDiff(deploymentName, catalog.getPlan(planId)))
-        .then(diff => utils.hasChangesInForbiddenSections(diff))
-        .tap(() => logger.info(`Doing update for ${deploymentName} as there is no forbidden changes in manifest`))
-        .then(() => updateDeployment());
-    } else {
-      logger.info(`Doing update for ${deploymentName} even if forbidden changes exist in manifest`);
-      return updateDeployment();
-    }
+
+    return Promise.try(() => {
+      logger.info(`Forbidden Manifest flag set to ${allowForbiddenManifestChanges}`);
+      return eventmesh.apiServerClient.getResource({
+          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+          resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
+          resourceId: instanceId
+        })
+        .catch(errors.NotFound, () => undefined)
+        .then(resource => _.get(resource, 'spec.options'))
+        .then(resource => {
+          resourceDetails = resource;
+          if (resourceDetails === undefined) {
+            throw new errors.NotFound(`Resource details of service instance ${instanceId} not found in api server.`);
+          } else {
+            const planId = _.get(resourceDetails, 'plan_id');
+            plan = catalog.getPlan(planId);
+            context = _.get(resourceDetails, 'context');
+            if (allowForbiddenManifestChanges === false) {
+              const tenantInfo = _.pick(resourceDetails, ['context', 'organization_guid', 'space_guid']);
```

Comment:
getting only context is not enough? Do we need to get organization_guid and space_guid also?

Reply:
Legacy code :) Looking at DirectorService.generateManifest() seems not required.

Reply:
lets remove then

Reply:
Checked the code, should be removed. As you said, it's a legacy code.

Reply:
done